src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -169,6 +169,7 @@ public async Task OnConnectionAsync(ConnectionContext context)
                     {
                         selector = (sender, name) =>
                         {
+                            feature.HostName = name;
```

Comment:
Should write log the hostname when a single cert is used and not an SNI callback meaning `_serverCertificateSelector` is null? We could always pass in a ServerCertificateSelectionCallback and just have it return the `_serverCertificate` when no custom callback is configured.

Also, don't you love it when you thought you left a comment days ago, but you accidentally started a review you didn't finsih?

Reply:
> Should write log the hostname when a single cert is used and not an SNI callback meaning _serverCertificateSelector is null? We could always pass in a ServerCertificateSelectionCallback and just have it return the _serverCertificate when no custom callback is configured.

Where do I get that information? Is it on SSlStream?

Reply:
> Where do I get that information? Is it on SSlStream?

I don't know. I don't think so. That's why I suggested always using a ServerCertificateSelectionCallback, so the hostname could be captured inside the callback.